cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/control/CheckStateListener.java

```diff
@@ -236,46 +236,6 @@ private void checkPageComplete() {
         }
     }
 
-    /**
-     * Performs an intelligent check logic such that the same element in different paths will be checked
-     * simultaneously, parents will be unselected if at least one child is not selected, and parents will be
-     * automatically selected if all children of the parent are selected
-     * @param event
-     *            triggering {@link CheckStateChangedEvent}
-     * @param packageSelector
-     *            current {@link CheckboxTreeViewer} for the package selection
-     */
-    private void performCheckLogic(CheckStateChangedEvent event, CheckboxTreeViewer packageSelector) {
-
-        SelectIncrementContentProvider cp = (SelectIncrementContentProvider) packageSelector.getContentProvider();
```

Comment:
In my opinion, we should not move this method to `SelectFileLabelProvider` because that class is only related to labels. Also the JavaDoc of `CheckStateListener` specifies the following:

> This CheckStateListener provides the check / uncheck of the increments list and the generationresource tree and synchronizes both in order to get a consistent view

This method should stay here.